### PR TITLE
GODRIVER-1958 Rename ServerID to ServiceID

### PR DESCRIPTION
**Pull request by divjotarora**
This change is to keep us in sync with a recent change in the LB spec.

### CODE REVIEW

File: event/monitoring.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ type CommandStartedEvent struct {
 	CommandName  string
 	RequestID    int64
 	ConnectionID string
-	ServerID     *primitive.ObjectID // Only set for servers that are deployed behind a load balancer.
+	ServiceID    *primitive.ObjectID // Only set for servers that are deployed behind a load balancer.
```
Comment (benjirewis) [inline event/monitoring.go:25]: Should any of these comments referencing "servers" now be "services" instead?

**Review by benjirewis — COMMENTED** [review]
Just to clarify, what services are possible? mongos, Atlas Proxy, Elastic front end?

File: event/monitoring.go (hunk 1 above)
Comment (divjotarora) [reply event/monitoring.go:25]: The wording is a little tricky here. I don't want to just change everything to "services" but the existing wording isn't great either. I updated the docs to more accurately describe what this field represents.

**Review by divjotarora — COMMENTED** [review]
Yes, those will be the main services that can run behind an LB.

File: x/mongo/driver/topology/connection_options.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ var DefaultDialer Dialer = &net.Dialer{}
 type Handshaker = driver.Handshaker
 
 // generationNumberFn is a callback type used by a connection to fetch its generation number given its server ID.
```
Comment (iwysiu) [inline x/mongo/driver/topology/connection_options.go:39]: nit: server ID -> serviceID

File: x/mongo/driver/topology/pool_generation_counter.go [hunk 3]
```diff
@@ -49,83 +49,83 @@ func (p *poolGenerationMap) disconnect() {
 
 // addConnection increments the connection count for the generation associated with the given server ID and returns the
```
Comment (iwysiu) [inline x/mongo/driver/topology/pool_generation_counter.go:50]: nit: server ID -> serviceID

File: x/mongo/driver/topology/server.go [hunk 4]
```diff
@@ -273,15 +273,15 @@ func (s *Server) Connection(ctx context.Context) (driver.Connection, error) {
 }
 
 // ProcessHandshakeError implements SDAM error handling for errors that occur before a connection finishes handshaking.
-func (s *Server) ProcessHandshakeError(err error, startingGenerationNumber uint64, serverID *primitive.ObjectID) {
+func (s *Server) ProcessHandshakeError(err error, startingGenerationNumber uint64, serviceID *primitive.ObjectID) {
 	// Ignore the error if the server is behind a load balancer but the server ID is unknown. This indicates that the
```
Comment (iwysiu) [inline x/mongo/driver/topology/server.go:277]: nit: server ID -> serviceID

**Review by benjirewis — APPROVED** [review]
Sounds good. LGTM!

**Review by iwysiu — APPROVED** [review]
I realize that serviceID is technically a server ID, so if the wording is deliberate it's fine to leave it

File: x/mongo/driver/topology/connection_options.go (hunk 2 above)
Comment (divjotarora) [reply x/mongo/driver/topology/connection_options.go:39]: Done.

File: x/mongo/driver/topology/pool_generation_counter.go (hunk 3 above)
Comment (divjotarora) [reply x/mongo/driver/topology/pool_generation_counter.go:50]: Done here and elsewhere in this file.

File: x/mongo/driver/topology/server.go (hunk 4 above)
Comment (divjotarora) [reply x/mongo/driver/topology/server.go:277]: Done here and elsewhere in this file.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**